tests/scripts/generate_tls13_compat_tests.py

```diff
@@ -511,6 +524,182 @@ def generate_hrr_compat_test(client=None, server=None,
+            ret += ["-groups {named_groups}".format(named_groups=named_groups)]
+
+        return ret
+
+
+class PossibleGroupGnuTLSCli(GnuTLSCli):
+    """
+    Generate PSK test command for GnuTLS Client
+    """
+    def cmd(self):
+        psk_identities = self._psk_identities or PSK_DEFAULT_IDENTITIES.values()
+        identity, psk = list(psk_identities)[0]
+
+        def update_priority_string_list(items, map_table):
+            for item in items:
+                for i in map_table[item]:
+                    yield '+' + i
+
+        priority_string_list = ['NORMAL', '-VERS-ALL', '-KX-ALL', '+ECDHE-PSK', '+DHE-PSK', '+PSK',
+                                '+VERS-TLS1.3', '-GROUP-ALL']
+        if self._named_groups:
+            priority_string_list.extend(update_priority_string_list(
+                self._named_groups, self.NAMED_GROUP))
+        ret = ret = super().pre_cmd() + ['--priority', ':'.join(priority_string_list),
+                                         '--pskusername', identity, '--pskkey', psk, 'localhost']
+        return ret
+
+
+class PossibleGroupMbedTLSServ(MbedTLSServ):
+    """
+    Generate PSK test command for MbedTLS server
+    """
+    def cmd(self):
+        _psk_identities = self._psk_identities or PSK_DEFAULT_IDENTITIES.values()
+        psk_list = []
+        for psk in _psk_identities:
+            psk_list.extend(psk)
+        ret = super().pre_cmd() + ['force_version=tls13',
+                                   'tls13_kex_modes=psk_ephemeral', 'debug_level=5']
+        if psk_list:
+            ret.append('psk_list={}'.format(','.join(psk_list)))
+        if self._named_groups:
+            named_groups = ','.join(self._named_groups)
+            ret += ["curves={named_groups}".format(named_groups=named_groups)]
+        return ret
+
+    def pre_checks(self):
+        configs = ['requires_all_configs_enabled', 'MBEDTLS_SSL_PROTO_TLS1_3',
+                   'MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE', 'MBEDTLS_SSL_SRV_C', 'MBEDTLS_DEBUG_C']
+        some_psk_enabled = ['requires_any_configs_enabled',
+                            'MBEDTLS_KEY_EXCHANGE_PSK_ENABLED',
+                            'MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED',
+                            'MBEDTLS_KEY_EXCHANGE_DHE_PSK_ENABLED',
+                            'MBEDTLS_KEY_EXCHANGE_ECDHE_PSK_ENABLED']
+        some_ephemeral_enabled = ['requires_any_configs_enabled',
+                                  'MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED',
+                                  'MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED',
+                                  'MBEDTLS_KEY_EXCHANGE_ECDHE_PSK_ENABLED']
+        return [' '.join(configs), ' '.join(some_psk_enabled), ' '.join(some_ephemeral_enabled)]
+
+    def post_checks(self):
+        return ['-S "key exchange mode: psk$"', '-s "key exchange mode: psk_ephemeral"',
+                '-S "key exchange mode: ephemeral"']
+
+
+POSSIBALEGROUP_SERVER_CLASSES = {'mbedTLS': PossibleGroupMbedTLSServ}
```

Comment:
```suggestion
PSK_EPHEMERAL_SERVER_CLASSES = {'mbedTLS': PossibleGroupMbedTLSServ}
```
Otherwise as for the HRR tests why do not we have: m->O, m->G, O->m, G->m and m->m ?

Reply:
Will add it